Project1/Party.cpp

```diff
@@ -0,0 +1,85 @@
+#include "Party.h"
+
+#include <utility>
+
+Party::Party() 
+= default;
+
+Party::Party(std::string name) : PartyBase(name)
+{}
+
+Party::Party(std::string name, std::vector<UnitData> unit_data) : PartyBase(name, unit_data)
+{}
+
+Party::Party(PartyData data) : PartyBase(data)
+{}
+
+void Party::startTurn(PartyBase& turn_party) {
+	if (this == &turn_party) {
+		for (auto& unit : _units) {
+			unit->refresh();
+		}
+	}
+	else {
+		for (auto& unit : _units) {
+			unit->newTurn();
+		}
+	}
+}
+void Party::insertUnit(UnitPtr& unit) {
```

Comment:
You don't have to reimplement any of these. You inherited PartyBase publicly, so everyone can call public PartyBase methods on you.

Reply:
InsertUnit was protected, should I make it public?